lib/importers/assigned_article_importer.rb

```diff
@@ -6,14 +6,18 @@ class AssignedArticleImporter
   ################
 
   def self.import_articles_for_assignments
-    titles = Assignment.where(article_id: nil).pluck(:article_title).uniq
-    ArticleImporter.import_articles_by_title(titles)
-    new_article_titles = Article.where(namespace: 0, title: titles).pluck(:title)
-    Assignment.where(article_title: new_article_titles).each do |assignment|
-      article = Article.find_by(title: assignment.article_title)
-      next unless article.title == assignment.article_title # guard against case variants
-      assignment.article_id = article.id
-      assignment.save
+    assignments_missing_articles = Assignment.where(article_id: nil)
+    assignments_missing_articles.group_by(&:wiki).each do |wiki, assignments|
+      missing_titles = assignments.map(&:article_title).uniq
+      ArticleImporter.new(wiki).import_articles_by_title(missing_titles)
+      # FIXME: How are these "new"?
```

Comment:
They are 'new' as in 'newly imported to the dashboard'. These are all the new Article records for Assignments that previously on had a title but weren't linked to an Article.